### DISCUSSION TITLE
test-bot fails because dependency cannot be installed from bottle

### DISCUSSION THREAD

**Discussion by nicholasdille** [Homebrew/discussions#823] [Tap maintenance and Homebrew development] [answered]
Hi everyone!

I hope this is the correct place for my issue. If not please let me know what to do :-)

I have a [formula for slirp4netns](https://github.com/nicholasdille/homebrew-tap/blob/main/Formula/slirp4netns.rb) which depends on `libslirp`. The repository was created with `brew tap-new` and executed the provided GitHb Actions.

When `test.yml` runs on this formula, it fails when running `brew install --only-dependencies --verbose --build-bottle nicholasdille/tap/slirp4netns` as part of `brew test-bot --only-formulae`. The output is very length so I only included the last few lines:

```
==> Downloading https://gitlab.freedesktop.org/slirp/libslirp/-/archive/v4.4.0/libslirp-v4.4.0.tar.gz
/usr/bin/curl --disable --globoff --show-error --user-agent Linuxbrew/3.0.1-67-ga946cde\ \(Linux\;\ x86_64\ Ubuntu\ 20.04.2\ LTS\)\ curl/7.68.0 --header Accept-Language:\ en --retry 3 --location --range 0-1 --dump-header - --write-out \%\{http_code\} --output /dev/null https://gitlab.freedesktop.org/slirp/libslirp/-/archive/v4.4.0/libslirp-v4.4.0.tar.gz
/usr/bin/curl --disable --globoff --show-error --user-agent Linuxbrew/3.0.1-67-ga946cde\ \(Linux\;\ x86_64\ Ubuntu\ 20.04.2\ LTS\)\ curl/7.68.0 --header Accept-Language:\ en --fail --retry 3 --location --remote-time --continue-at 0 --output /home/dillen/.cache/Homebrew/downloads/927a513c45b0b2cfdbb0b20bfb661f44367b8ee8c4a03eba91fda7c238e02e2a--libslirp-v4.4.0.tar.gz.incomplete https://gitlab.freedesktop.org/slirp/libslirp/-/archive/v4.4.0/libslirp-v4.4.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  125k    0  125k    0     0   183k      0 --:--:-- --:--:-- --:--:--  183k
==> Verifying checksum for '927a513c45b0b2cfdbb0b20bfb661f44367b8ee8c4a03eba91fda7c238e02e2a--libslirp-v4.4.0.tar.gz'
Error: The following formula cannot be installed from bottle and must be
built from source.
  libslirp
```

This can be reproduced with the following steps:

1. `brew tap nicholasdille/tap`

1. `brew install --only-dependencies --verbose --build-bottle nicholasdille/tap/slirp4netns`

When I install `libslirp` explicitly first and run the above command it works. It looks like some internal logic is preventing `libslirp` to be built from source when building a bottle.

What can I do about this?

Thanks in advance
Nicholas

**Comment by SMillerDev** [comment]
I believe @homebrew/Linux used to have the same issue and resolved it.

**Reply by iMichka** [reply]
The solution is to have a bottle, that's all :)
I triggered a build for it here: https://github.com/Homebrew/linuxbrew-core/actions/runs/571021043

**Comment by iMichka** [comment, marked as answer]
The bottle is built and uploaded, can you try again?

**Reply by nicholasdille** [reply]
Awesome it worked :-) Thanks for your help!

Is there an official way to request a bottle in case this comes up again?

**Reply by carlocab** [reply]
I think the official way is to catch @iMichka in a good mood.

**Reply by nicholasdille** [reply]
Is there an official way to determine @iMichka s mood? ;-)

**Reply by iMichka** [reply]
I'll write an API for that :D

Ha ha, no, more seriously, you can open an issue at https://github.com/Homebrew/linuxbrew-core/issues and let us know.

In the future we plan to make sure there is always a bottle, when linuxbrew-core is merged into homebrew-core; that'll happen later this year I hope.
